lib/pacemaker/pcmk_fence.c

```diff
@@ -156,7 +161,7 @@ pcmk__fence_action(stonith_t *st, const char *target, const char *action,
 
     free(async_fence_data.name);
 
-    return pcmk_legacy2rc(async_fence_data.rc);
+    return stonith__result2rc(&async_fence_data.result);
```

Comment:
nothing that would build up a leak but valgrind might complain if we don't reset the result here.

Reply:
good idea